thirdparty/rapidjson-48fbd8cd202ca54031fe799db2ad44ffa8e77c13/include/rapidjson/document.h

```diff
@@ -1236,7 +1236,7 @@ class GenericValue {
 
             // Use static buffer and placement-new to prevent destruction
             static char buffer[sizeof(GenericValue)];
-            return *new (buffer) GenericValue();
+            return *new (buffer) GenericValue();  // NOLINT(clang-analyzer-cplusplus.PlacementNew)
```

Comment:
Can we exclude the thirdparty directory instead?

Reply:
Rapidjson is set to be a system include so it should be exluded by clang-tidy. It does work for every other thirdparty, but somehow when a file uses this inlude clang-tidy still throws a warning about this. Unfortunately I could not find another way of excluding it and it may be due to a clang-tidy-14 bug because when I checked the same files that had this error with clang-tidy-10 I did not get any warnings.